Don't kill owned road entities (leashed / named)

BlockyTheDev opened the pull request:
## Overview

Fixes #3824

## Description
Fixes a bug, where leashed entities got killed on the road, with "KILL_ROAD_MOBS" set to true and "KILL_OWNED_ROAD_MOBS" set to false. This PR restores the behaviour described in the settings comment at the value "KILL_OWNED_ROAD_MOBS".

## Submitter Checklist
<!-- Make sure you have completed the following steps (put an "X" between of brackets): -->
- [ ] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [ ] Ensure that the pull request title represents the desired changelog entry.
- [ ] New public fields and methods are annotated with `@since TODO`.
- [ ] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md).

PierreSchwang commented:
Would be great if you could adress the 2nd part of that issue as well ("Mobs not being removed when leashed **or named**") - Otherwise that issue would only be partly solved

BlockyTheDev commented:
> Would be great if you could adress the 2nd part of that issue as well ("Mobs not being removed when leashed **or named**") - Otherwise that issue would only be partly solved

Ok. I will do it tomorrow. I didn't added the name thing, because I oriented me on the settings comment. Should I add the thing with the name to the settings comment, too?

BlockyTheDev commented:
Or should there be a extra setting for named items?

The problem, I am seeing, when the named thing is the same setting like the leashed and ridden one, is primarily on servers where nametags are available in a large number. So players can bring very much entities to the road. This is not always wanted, because players could bypass things like plot caps with things like villagers by adding a nametag to them and moving them to the road and switching them after trading, so they can have much more, as planned.

I hope I wrote it in an understandable way.

BlockyTheDev commented:
> > Or should there be a extra setting for named items?
> > The problem, I am seeing, when the named thing is the same setting like the leashed and ridden one, is primarily on servers where nametags are available in a large number. So players can bring very much entities to the road. This is not always wanted, because players could bypass things like plot caps with things like villagers by adding a nametag to them and moving them to the road and switching them after trading, so they can have much more, as planned.
> > I hope I wrote it in an understandable way.
> 
> Good point - I'd say you are free to decide that. You can add a new config option for that or keep it as it is

I added a setting to kill named mobs. In addition, I added a missing check on shulkers, where shulkers are removed when they are leashed but the setting `KILL_OWNED_ROAD_MOBS` is set to false.

dordsor21 commented on Core/src/main/java/com/plotsquared/core/configuration/Settings.java:
```diff
@@ -759,6 +759,8 @@ public static final class Enabled_Components { // Group the following values int
         @Comment("Also kill any road mobs that are being ridden, or are leashed")
         public static boolean
                 KILL_OWNED_ROAD_MOBS = false;
+        @Comment("Also kill any road mobs that are being named")
```
```suggestion
        @Comment("Also kill any road mobs that are named")
```
